// @mui/x-data-grid-generator/legacy/renderer/renderEditRateType.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { GridEvents } from '@mui/x-data-grid-pro';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import { RATE_TYPE_OPTIONS } from '../services/static-data';
import { jsx as _jsx } from "react/jsx-runtime";

function EditRateType(props) {
  var id = props.id,
      value = props.value,
      api = props.api,
      field = props.field;

  var handleChange = function handleChange(event) {
    api.setEditCellValue({
      id: id,
      field: field,
      value: event.target.value
    }, event);
    api.commitCellChange({
      id: id,
      field: field
    });
    api.setCellMode(id, field, 'view');

    if (event.key) {
      // TODO v6: remove once we stop ignoring events fired from portals
      var params = api.getCellParams(id, field);
      api.publishEvent(GridEvents.cellNavigationKeyDown, params, event);
    }
  };

  var handleClose = function handleClose(event, reason) {
    if (reason === 'backdropClick') {
      api.setCellMode(id, field, 'view');
    }
  };

  return /*#__PURE__*/_jsx(Select, {
    value: value,
    onChange: handleChange,
    MenuProps: {
      onClose: handleClose
    },
    sx: {
      height: 1,
      '& .MuiSelect-select': {
        display: 'flex',
        alignItems: 'center',
        pl: 1
      }
    },
    autoFocus: true,
    fullWidth: true,
    open: true,
    children: RATE_TYPE_OPTIONS.map(function (option) {
      return /*#__PURE__*/_jsx(MenuItem, {
        value: option,
        children: option
      }, option);
    })
  });
}

export function renderEditRateType(params) {
  return /*#__PURE__*/_jsx(EditRateType, _extends({}, params));
}
