// @mui/x-data-grid-generator/renderer/renderEditTaxCode.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { GridEvents } from '@mui/x-data-grid-pro';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import { TAXCODE_OPTIONS } from '../services/static-data';
import { jsx as _jsx } from "react/jsx-runtime";

function EditTaxCode(props) {
  const {
    id,
    value,
    api,
    field
  } = props;

  const handleChange = event => {
    api.setEditCellValue({
      id,
      field,
      value: event.target.value
    }, event);
    api.commitCellChange({
      id,
      field
    });
    api.setCellMode(id, field, 'view');

    if (event.key) {
      // TODO v6: remove once we stop ignoring events fired from portals
      const params = api.getCellParams(id, field);
      api.publishEvent(GridEvents.cellNavigationKeyDown, params, event);
    }
  };

  const handleClose = (event, reason) => {
    if (reason === 'backdropClick') {
      api.setCellMode(id, field, 'view');
    }
  };

  return /*#__PURE__*/_jsx(Select, {
    value: value,
    onChange: handleChange,
    MenuProps: {
      onClose: handleClose
    },
    sx: {
      height: 1,
      '& .MuiSelect-select': {
        display: 'flex',
        alignItems: 'center',
        pl: 1
      }
    },
    autoFocus: true,
    fullWidth: true,
    open: true,
    children: TAXCODE_OPTIONS.map(option => {
      return /*#__PURE__*/_jsx(MenuItem, {
        value: option,
        children: option
      }, option);
    })
  });
}

export function renderEditTaxCode(params) {
  return /*#__PURE__*/_jsx(EditTaxCode, _extends({}, params));
}
